capstone/src/main/java/com/google/sps/servlets/EditProfileServlet.java

```diff
@@ -88,8 +88,14 @@ private Entity getExistingUser(HttpServletResponse response, DatastoreService da
    * Creates updated user object to return.
    */
   private User getUpdatedUser(Entity entity, String first, String last, String email, String phone, 
-      ArrayList<String> interests) {
-    User user = User.fromEntity(entity);
+      ArrayList<String> interests, HttpServletResponse response) throws IOException {
+    User user;
+    try {
+      user = User.fromEntity(entity);
+    } catch (EntityNotFoundException e) {
+      ErrorHandler.sendError(response, "User entity error.");
+      return null;
+    }
```

Comment:
This is good, however I'm not a fan of the generic "User entity error". This is my fault since I told you to get rid of thr previous try/catch block, but could you add that back but instead throw a new EntityNotFoundException with a relevant message in getBadgeList and use that message from e here instead?

Reply:
Okay! Just to clarify, instead of putting "throws EntityNotFoundException" in getBadgeList() header, I should directly throw the error with a message in a catch block in getBadgeList()?